File: uniffi_bindgen/src/bindings/kotlin/templates/Int32Helper.kt
internal object FfiConverterI32 {
    fun lift(v: Int): Int {
        return v
    }

    fun read(buf: ByteBuffer): Int {
        return buf.getInt()
    }

    fun lower(v: Int): Int {
        return v
    }

    fun write(v: Int, buf: RustBufferBuilder) {
        buf.putInt(v)
    }
}

File: uniffi_bindgen/src/bindings/swift/templates/Int32Helper.swift
fileprivate struct FfiConverterInt32: FfiConverterPrimitive {
    typealias FfiType = Int32
    typealias SwiftType = Int32

    static func read(from buf: inout (data: Data, offset: Data.Index)) throws -> Int32 {
        return try lift(readInt(&buf))
    }

    static func write(_ value: Int32, into buf: inout [UInt8]) {
        writeInt(&buf, lower(value))
    }
}
